src/Response.php

```diff
@@ -33,10 +33,15 @@ public function __construct(DuplexStreamInterface $stream, $protocol, $version,
         $this->code = $code;
         $this->reasonPhrase = $reasonPhrase;
         $this->headers = $headers;
+        $normalizedHeaders = array_change_key_case($headers, CASE_LOWER);
 
-        $stream->on('data', array($this, 'handleData'));
-        $stream->on('error', array($this, 'handleError'));
-        $stream->on('end', array($this, 'handleEnd'));
+        if (isset($normalizedHeaders['transfer-encoding']) && strtolower($normalizedHeaders['transfer-encoding']) === 'chunked') {
+            $this->stream = new ChunkedStreamDecoder($stream);
+        }
```

Comment:
Technically, this introduces a BC break.

How can a consumer of this library tell if the `Transfer-Encoding` has already been processed?

Should we raise to `v0.5.0`?